lib/NativeCall.pm6

```diff
@@ -403,7 +390,7 @@ sub cglobal($libname, $symbol, $target-type) is export is rw {
     Proxy.new(
         FETCH => -> $ {
             nqp::nativecallglobal(
-                nqp::unbox_s(guess_library_name($libname)),
+                nqp::unbox_s(guess_library_name($libname).IO.absolute),
```

Comment:
guess_library_name returns something dlopen and co can use to load the right library. That why a path or a name is valid.

Also some things outside NC.pm use guess_library_name (it's not ideal, but it's the only way to avoid rewriting it when we want run time lib resolution see: #716 for more details) so it's not that easy to change it

For the missing test on win32/os X. I just did not have these platforms set up to write tests for it.